app/src/main/java/com/example/songme/data/source/OnDataCallback.kt

```diff
@@ -0,0 +1,8 @@
+package com.example.songme.data.source
+
+import com.example.songme.data.model.Track
+
+interface OnDataCallback {
+    fun onSuccess(data: List<Track>)
+    fun onFailed(exception: Exception)
+}
```

Comment:
sử dụng generic type đi ông 
```suggestion
interface OnDataLoadedCallback<T> {
    fun onSuccess(data: T)
    fun onFailed(exception: Exception)
}
```